Automattic/wp-calypso
# Framework: Remove ObserveWindowSizeMixin

In our continuous attempt to migrate away from `React.createClass`, we're removing mixins as much as possible.

Since #10105 landed, the only usage of `ObserveWindowSizeMixin` was in `HeaderCakeBack`, so it's the right time to migrate away from it. This PR removes the usage of `ObserveWindowSizeMixin` from `HeaderCakeBack` and then deletes the now unused mixin.

ES6-ification of the `HeaderCakeBack` and migration away from the deprecated `forceUpdate` is coming in another PR (#17533).

To test:
* Checkout this branch.
* Go to a theme page, for example http://calypso.localhost:3000/theme/apostrophe-2
* Resize your window to a mobile size (< 480 pixels)
* Verify the back button text is not visible.
* Resize your window to a large size again.
* Verify the back button text is visible.
